js/data/bucket/symbol_bucket.js

```diff
@@ -327,6 +331,12 @@ SymbolBucket.prototype.addFeature = function(lines, shapedText, shapedIcon, feat
                         iconBoxScale, iconPadding, iconAlongLine));
         }
     }
+
+    var addSymbolInstance = this.addSymbolInstance.bind(this);
+    this.symbolInstances.forEach(function(instance) {
+        addSymbolInstance(instance);
+    });
```

Comment:
I've found that bound functions are significantly slower than unbound ones. Because this is hot code, I recommend using plain ol `for` loop and calling `this.addSymbolInstance `